### DISCUSSION TITLE
Support MariaDB/Oracle DB type through Django?

### DISCUSSION THREAD

**Discussion by ArKam** [netbox-community/netbox#11896] [General]
Hi team,

I noticed that netbox is built with Django, which support any DB (MariaDB/Oracle/PostgreSQL) is there any particular reason why netbox do specify postgres DB usage? Any PostgreSQL specific feature leveraged?
I'm currently trying to harmonize our DB usage on various services through MariaDB and though about it for netbox, but I didn't find particular requirements for PostgreSQL so far.

Thanks!

**Comment by candlerb** [comment]
Netbox makes heavy use of Postgres-specific data types for IP addresses and MAC addresses; jsonb columns and queries for [custom fields](https://docs.netbox.dev/en/stable/customization/custom-fields/); array columns (e.g. for custom field choices and rack reservation units); probably other things that I haven't thought of

To use any other database you would have to fork Netbox and be left maintaining your own fork, and it will break in subtle and not-so-subtle ways.  You really don't want to do this.   IMO, if you can't use Postgres in your environment (not even a cloud-based Postgres like Amazon RDS) then you'd be better off not deploying Netbox.
